### PR TITLE
debian: move the packaging out into package/$id-$version_id

### PR DESCRIPTION
**Pull request by mvo5**
This moves the packaging into `package/$id-$version_id`. It also keeps a compatibility symlink to the current `./debian` dir based on the 16.04 packaging. This is to ensure that tools like `gbp` and `autopkgtest` continue working. Without a debian/ dir they fail.

### CODE REVIEW

File: generate-packaging-dir [hunk 1]
```diff
@@ -1,32 +1,9 @@
...
-)
+. /etc/os-release
 
-if [ ! -d .git ]; then
-    echo "not running in a git checkout, skipping"
-    exit 0
-fi
+# update symlink
+rm -f debian
+ln -s packaging/${ID}-${VERSION_ID} debian
```
Comment (zyga) [inline generate-packaging-dir]: I think we should only do this if we're on a debian-like system. Also note that sid has no VERSION_ID so I think we should check for it as well.

Comment (mvo5) [inline generate-packaging-dir]: I will add FIXMEs - as soon as we add debian or a non-debian system we can fix this script easily.

File: packaging/ubuntu-16.04/not-installed [hunk 2]
```diff
@@ -0,0 +1 @@
+debian/tmp/usr/bin/uboot-go
```
Comment (chipaca) [inline packaging/ubuntu-16.04/not-installed]: I didn't know about `not-installed`. Assuming it flags things to not warn about not installing, we might be able to drop a couple of `rm`'s from the `override_dh_install` rule

Comment (mvo5) [inline packaging/ubuntu-16.04/not-installed]: Very good point, I think we want a followup with that cleanup.

**Pull request merged**